statsmodels/stats/tests/test_cov_tools.py

```diff
@@ -0,0 +1,269 @@
+                 [1.601, 0.736, 0.599],
+                 [0.292, 0.155, 0.289],
+                 [7.020, 5.987, 7.403]])
+
+    z = (x - x.mean(0)) / x.std(0, ddof=1)  # not rounded as in z0
+    ct.mom4(z0)
+
+    def ztest_correlated(diff, var0, var1, covar, nobs):
+        std = np.sqrt((var0 + var1 - 2 * covar) / nobs)
+        statistic = diff / std
+        from scipy import stats
+        pvalue = stats.norm.sf(np.abs(statistic)) * 2
+        return statistic, pvalue
+
+    corr = np.corrcoef(z, rowvar=0)
+    #corr = z0.dot(z0) / len(z0)
+    # constrained correlation under corr_r[0, 1] = corr_r[0, 2]
+    # estimate: replace by simple average (OLS)
+    corr_r = corr.copy()
+    corr_r[0, 1] = (corr[0, 1] + corr[0, 2]) / 2
+    corr_r[1, 0] = corr_r[0, 2] = corr_r[2,0] = corr_r[0, 1]
+    diff = corr[0, 1] - corr[0, 2]
+
+    # two cases under normality and with empirical mom4
+    # using functions for individual correlation coefficients
+
+    # without normality assumption for 4th moments
+    m4 = ct.mom4(z, ddof=1).reshape(3,3,3,3)
+
+    i, j, k, h = 0, 1, 0, 1
+    g1 = cov_corr_coef(i, j, k, h, corr_r, m4)
+    assert_allclose(g1, 0.20200121658014503, rtol=1e-8)
+    res = 0.2020
+    assert_allclose(g1, res, atol=5e-4)
+
+    i, j, k, h = 0, 2, 0, 2;
+    g2 = cov_corr_coef(i, j, k, h, corr_r, m4)
+    0.10611046338328745
+    res = 0.1062
+    assert_allclose(g2, res, atol=5e-4)
+
+    i, j, k, h = 0, 1, 0, 2;
+    g3 = cov_corr_coef(i, j, k, h, corr_r, m4)
+    0.048605932563082987
+    res = 0.0486
+    assert_allclose(g3, res, atol=5e-4)
+
+    i, j, k, h = 1, 2, 1, 2;
+    g4 = cov_corr_coef(i, j, k, h, corr_r, m4)
+    0.051162679039107317
+
+    var0, var1, covar, nobs = g1, g2, g3, 20
+    zt, pv = ztest_correlated(diff, var0, var1, covar, nobs)
+    (-1.2978383834464489, 0.19434287689430729)
+    zt1 = -1.25   # difference is most likely rounding error, g3 differs a bit
+
+    # with normality assumption for 4th moments
+
+    i, j, k, h = 0, 1, 0, 1
+    g1 = ct.cov_corr_coef_normal(i, j, k, h, corr_r)
+    0.17012301158351617
+    res = 0.1701
+
+    i, j, k, h = 0, 2, 0, 2
+    g2 = ct.cov_corr_coef_normal(i, j, k, h, corr_r)
+    0.17012301158351617
+    res = 0.1701
+    i, j, k, h = 0, 1, 0, 2
+    g3 = ct.cov_corr_coef_normal(i, j, k, h, corr_r)
+    0.13832670991613805
+    res = 0.1383
+
+    # normal theory variance of correlation coefficient is
+    cr0 = (1-corr_r**2)**2
+    assert_allclose(cr0[1,0], g1, rtol=1e-14)
+    assert_allclose(cr0[1,0], g2, rtol=1e-14)
+
+    cr0low = ct.veclow((1-corr**2)**2)
+    crn = [ct.cov_corr_coef_normal(0, 1, 0, 1, corr),
+           ct.cov_corr_coef_normal(0, 2, 0, 2, corr),
+           ct.cov_corr_coef_normal(1, 2, 1, 2, corr)]
+    assert_allclose(cr0low, crn, atol=1e-14)
+
+    var0, var1, covar, nobs = g1, g2, g3, 20
+    ztest_correlated(diff, var0, var1, covar, nobs)
+    (-2.3642128782748038, 0.018068426888646225)
+    zt1 = -2.30
```

Comment:
the ztest is an application in Steiger and Hackstian. Not converted to unit tests yet, and no other support for hypothesis tests yet.